extra/html-extra/HtmlAttributes.php

```diff
@@ -0,0 +1,221 @@
+
+            $attributeGroupCount++;
+
+            // Skip empty attributes
+            // Return early if no attributes are provided
+            // This could be false or null when using the twig ternary operator
+            if (!$attributes) {
+                continue;
+            }
+
+            if (!is_iterable($attributes)) {
+                throw new RuntimeError(sprintf('"%s" only works with mappings or "Traversable", got "%s" for argument %d.', self::class, \gettype($attributes), $attributeGroupCount));
+            }
+
+            // Alternative to is_iterable check above, cast the attributes to an array
+            // This would produce weird results but would not throw an error
+//            $attributes = (array)$attributes;
+
+            // data and aria arrays are expanded into data-* and aria-* attributes
+            $expanded = [];
+            foreach ($attributes as $key => $value) {
+                if (in_array($key, ['data', 'aria'])) {
+                    $value = (array)$value;
+                    foreach ($value as $k => $v) {
+                        $k = $key . '-' . $k;
+                        $expanded[$k] = $v;
+                    }
+                    continue;
+                }
+                $expanded[$key] = $value;
+            }
+
+            // Reset the attributes array to the flattened version
+            $attributes = $expanded;
+
+            foreach ($attributes as $key => $value) {
+
+                // Treat class and data-controller attributes as arrays
+                if (in_array($key, [
+                    'class',
+                    'data-controller',
+                    'data-action',
+                    'data-targets',
+                ])) {
+                    if (!array_key_exists($key, $result)) {
+                        $result[$key] = [];
+                    }
+                    $value = (array)$value;
+                    foreach ($value as $k => $v) {
+                        if (is_int($k)) {
+                            $classes = explode(' ', $v);
+                            foreach ($classes as $class) {
+                                $result[$key][$class] = true;
+                            }
+                        } else {
+                            $classes = explode(' ', $k);
+                            foreach ($classes as $class) {
+                                $result[$key][$class] = $v;
+                            }
+                        }
+                    }
+                    continue;
+                }
+
+                if ($key === 'style') {
+                    if (!array_key_exists('style', $result)) {
+                        $result['style'] = [];
+                    }
+                    $value = (array)$value;
+                    foreach ($value as $k => $v) {
+                        if (is_int($k)) {
+                            $styles = array_filter(explode(';', $v));
```

Comment:
this is broken if a CSS string contains a `;`. You cannot perform CSS parsing based on naive `explode` calls. The CSS syntax is not meant to allow that.